packages/worker/src/worker.ts

```diff
@@ -1644,13 +1734,11 @@ export class Worker {
         );
       } finally {
         Runtime.instance().deregisterShutdownSignalCallback(shutdownCallback);
+
+        await this.nativeWorker.finalizeShutdown();
       }
```

Comment:
Can Rust hang here with corrupted state?

Reply:
I think there's a very slight possibly that it may happen, but in most "corrupted internal state" situations, Core will just panic, which will terminate the process anyway.

On the opposite, there's a non-negligeable probability that not calling `finalizeShutdown()` might result in difficulties shuting down the process.